### ISSUE TITLE
Incorrect hover documentation for `syntax "solve" : term` 

### ISSUE THREAD

**Issue by lecopivo** [leanprover/lean4#2449]
I have defined a custom notation for terms with the keyword `solve`.  When hovering over such term I'm getting documentation of already existing `solve` tactic and not the documentation for the `solve` term.

mwe
```
import Lean
open Lean Parser Elab Term 

def solveFun (f : α → Prop) : α := sorry

/-- solve term -/ 
elab "solve" xs:funBinder* ", " b:term : term => do
  let e ← elabTerm (← `(solveFun fun $xs* => $b)) none
  return e

#check solve x, x = 10  
``` 
Hovering over `solve` in the last command displays ``Similar to `first`, but succeeds only if one the given tactics solves the current goal.`` I'm expecting to see `solve term`

Tested on 

leanprover/lean4:nightly-2023-08-18
and the behavior is the same in Emacs as in VS Code

**Comment by tydeu** [comment]
Note that what you get here depends on where you hover. Highlighting the keyword (1) gives the tactic docs, but highlighting the middle of the term (2) gives the term docs
````lean
#check solve x, x = 10  
   --    ^1    ^2
```

**Comment by lecopivo** [comment]
Odd, but still that is not expected behavior, right?

**Comment by semorrison** [comment]
Was this fixed by #2586?

**Comment by semorrison** [comment]
Yes!
